clicommand/util_keygen.go

```diff
@@ -0,0 +1,111 @@
+package clicommand
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/buildkite/agent/v3/internal/jwkutil"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/urfave/cli"
+	"golang.org/x/exp/slices"
+)
+
+type KeygenConfig struct {
+	Alg                  string `cli:"alg" validate:"required"`
+	KeyID                string `cli:"key-id" validate:"required"`
+	PrivateKeyFilename   string `cli:"private-key-filename" normalize:"filepath"`
+	PublicKeysetFilename string `cli:"public-keyset-filename" normalize:"filepath"`
+
+	LogLevel string `cli:"log-level"`
+	Debug    bool   `cli:"debug"`
```

Comment:
I think the usual set of
```
		// Global flags
		NoColorFlag,
		DebugFlag,
		LogLevelFlag,
		ExperimentsFlag,
		ProfileFlag,
```
should be copy pasted here. I know I said I have a way of using type constraints to eliminate this boilerplate. I haven't gotten around to implement it. Feel free to take it off my hands.

Reply:
I know most of them don't make sense yet, but it will be easier to remove the boilerplate if every command has the same boilerplate.